### PR TITLE
64-wholeBrain-LargeUNEST-segmentation

### PR DESCRIPTION
**Pull request by tangy5**
Description
The whole brain Segmentation Bundle is making and to be added to the model zoo.
To added to next release of model-zoo

Status
Made and pend to test more cases.
wholebrain

Screenshot from 2022-08-10 22-57-43

- [x] Making the bundle scripts
- [x] Upload the model checkpoint
- [x] Check code format
- [x] Check customized bundle compatible and CI test

### CODE REVIEW

File: models/renalStructures_UNEST_segmentation/docs/README.md [hunk 1]
```diff
@@ -1,7 +1,13 @@
 # Description
 A pre-trained model for inferencing volumetric (3D) kidney substructures segmentation from contrast-enhanced CT images (Arterial/Portal Venous Phase).
+<<<<<<< HEAD
 A tutorial and release of model for kidney cortex, medulla and collecting system segmentation.
 
+=======
+
+A tutorial and release of model for kidney cortex, medulla and collecting system segmentation.
+
+>>>>>>> db7fd87494d8c7ad9b766bdf6e4010c57a205c79
```
Comment (Nic-Ma) [inline models/renalStructures_UNEST_segmentation/docs/README.md]: Please resolve the conflicts here, only keep the right doc.

Thanks.

File: models/renalStructures_UNEST_segmentation/configs/inference.json [hunk 2]
```diff
@@ -112,7 +112,7 @@
             "_target_": "CheckpointLoader",
             "load_path": "$@bundle_root + '/models/model.pt'",
             "load_dict": {
-                "state_dict": "@network"
+                "model": "@network"
```
Comment (Nic-Ma) [inline models/renalStructures_UNEST_segmentation/configs/inference.json:115]: If you update the bundle content, please update the version in the `metadata.json`.

Thanks.

File: models/wholeBrainSeg_Large_UNEST_segmentation/docs/README.md [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+|  108 :  Left-PP----planum-polare  |  109 :  Right-PrG---precentral-gyrus  |  110 :  Left-PrG---precentral-gyrus  |  111 :  Right-PT----planum-temporale  |
+|  112 :  Left-PT----planum-temporale  |  113 :  Right-SCA---subcallosal-area  |  114 :  Left-SCA---subcallosal-area  |  115 :  Right-SFG---superior-frontal-gyrus  |
+|  116 :  Left-SFG---superior-frontal-gyrus  |  117 :  Right-SMC---supplementary-motor-cortex  |  118 :  Left-SMC---supplementary-motor-cortex  |  119 :  Right-SMG---supramarginal-gyrus  |
+|  120 :  Left-SMG---supramarginal-gyrus  |  121 :  Right-SOG---superior-occipital-gyrus  |  122 :  Left-SOG---superior-occipital-gyrus  |  123 :  Right-SPL---superior-parietal-lobule  |
+|  124 :  Left-SPL---superior-parietal-lobule  |  125 :  Right-STG---superior-temporal-gyrus  |  126 :  Left-STG---superior-temporal-gyrus  |  127 :  Right-TMP---temporal-pole  |
+|  128 :  Left-TMP---temporal-pole  |  129 :  Right-TrIFG-triangular-part-of-the-IFG  |  130 :  Left-TrIFG-triangular-part-of-the-IFG  |  131 :  Right-TTG---transverse-temporal-gyrus  |
+|  132 :  Left-TTG---transverse-temporal-gyrus  |
+
+
+## Bundle Integration in MONAILable
```
Comment (Nic-Ma) [inline models/wholeBrainSeg_Large_UNEST_segmentation/docs/README.md]: typo, MONAI Label.

Thanks.

File: models/wholeBrainSeg_Large_UNEST_segmentation/docs/README.md [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+## Training configuration
+The training and inference was performed with at least one 24GB-memory GPU.
+
+Actual Model Input: 96 x 96 x 96
+
+## Input and output formats
+Input: 1 channel T1w MRI image in MNI305 Space.
+
+
+## Performance
```
Comment (Nic-Ma) [inline models/wholeBrainSeg_Large_UNEST_segmentation/docs/README.md]: If no performance info, please delete this part.
Or I would suggest to add some inference speed and memory requirements data.

Thanks.

File: models/wholeBrainSeg_Large_UNEST_segmentation/docs/README.md [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+
+Yinchi Zhou (yinchi.zhou@vanderbilt.edu) | Yucheng Tang (yuchengt@nvidia.com)
+
+<p align="center">
+-------------------------------------------------------------------------------------
+</p>
+
+<!-- ![](./demo.png) <br> -->
+<p align="center">
+<img src="./demo.png" width="800" height="456">
```
Comment (Nic-Ma) [inline models/wholeBrainSeg_Large_UNEST_segmentation/docs/README.md]: To insert image in the markdown document, you can refer to:
https://github.com/Project-MONAI/MONAI/blob/dev/docs/source/highlights.md?plain=1#L87

Thanks.

File: models/wholeBrainSeg_Large_UNEST_segmentation/docs/wholeBrainClassesList.txt [hunk 6]
```diff
@@ -0,0 +1,4 @@
+Third Party Licenses
```
Comment (Nic-Ma) [inline models/wholeBrainSeg_Large_UNEST_segmentation/docs/wholeBrainClassesList.txt]: What's this file used for?

Thanks.

File: models/wholeBrainSeg_Large_UNEST_segmentation/eval/schema.json [hunk 7]
```diff
@@ -0,0 +1,241 @@
+{
```
Comment (Nic-Ma) [inline models/wholeBrainSeg_Large_UNEST_segmentation/eval/schema.json]: Please remove the `eval` folder in the PR.

Thanks.

File: models/wholeBrainSeg_Large_UNEST_segmentation/docs/license.txt [hunk 8]
```diff
@@ -0,0 +1,4 @@
+Third Party Licenses
+-----------------------------------------------------------------------
+
+/*********************************************************************/
```
Comment (Nic-Ma) [inline models/wholeBrainSeg_Large_UNEST_segmentation/docs/license.txt:4]: Do you have any license information for the dataset used in this PR? You can refer to:
https://github.com/Project-MONAI/model-zoo/blob/dev/models/spleen_ct_segmentation/docs/license.txt#L5

Thanks.

File: models/renalStructures_UNEST_segmentation/docs/README.md (hunk 1 above)
Comment (tangy5) [reply models/renalStructures_UNEST_segmentation/docs/README.md]: Deleted, thanks!

File: models/renalStructures_UNEST_segmentation/configs/inference.json (hunk 2 above)
Comment (tangy5) [reply models/renalStructures_UNEST_segmentation/configs/inference.json:115]: Done

File: models/wholeBrainSeg_Large_UNEST_segmentation/docs/README.md (hunk 3 above)
Comment (tangy5) [reply models/wholeBrainSeg_Large_UNEST_segmentation/docs/README.md]: Fixed.

File: models/wholeBrainSeg_Large_UNEST_segmentation/docs/README.md (hunk 4 above)
Comment (tangy5) [reply models/wholeBrainSeg_Large_UNEST_segmentation/docs/README.md]: Deleted in the README file, thanks

File: models/wholeBrainSeg_Large_UNEST_segmentation/docs/README.md (hunk 5 above)
Comment (tangy5) [reply models/wholeBrainSeg_Large_UNEST_segmentation/docs/README.md]: Sure, changed back to the format.

File: models/wholeBrainSeg_Large_UNEST_segmentation/docs/wholeBrainClassesList.txt (hunk 6 above)
Comment (tangy5) [reply models/wholeBrainSeg_Large_UNEST_segmentation/docs/wholeBrainClassesList.txt]: It was a list of brain ROIs used for training, removed.

File: models/wholeBrainSeg_Large_UNEST_segmentation/eval/schema.json (hunk 7 above)
Comment (tangy5) [reply models/wholeBrainSeg_Large_UNEST_segmentation/eval/schema.json]: Done

File: models/wholeBrainSeg_Large_UNEST_segmentation/docs/license.txt (hunk 8 above)
Comment (tangy5) [reply models/wholeBrainSeg_Large_UNEST_segmentation/docs/license.txt:4]: No, the dataset for interence and training are public, but the labels are partly internal. License is good here. Thanks for confirming.

Comment (Nic-Ma) [reply models/wholeBrainSeg_Large_UNEST_segmentation/docs/license.txt:4]: I think even for the public dataset, it may have the license? like Decathlon dataset.
Could you please help confirm it?

Thanks.

File: models/renalStructures_UNEST_segmentation/docs/README.md [hunk 9]
```diff
@@ -1,7 +1,9 @@
 # Description
 A pre-trained model for inferencing volumetric (3D) kidney substructures segmentation from contrast-enhanced CT images (Arterial/Portal Venous Phase).
+
 A tutorial and release of model for kidney cortex, medulla and collecting system segmentation.
 
+>>>>>>> db7fd87494d8c7ad9b766bdf6e4010c57a205c79
```
Comment (Nic-Ma) [inline models/renalStructures_UNEST_segmentation/docs/README.md]: Please remove the git conflicts comment.

Thanks.

File: models/renalStructures_UNEST_segmentation/docs/README.md [hunk 10]
```diff
@@ -1,7 +1,9 @@
 # Description
 A pre-trained model for inferencing volumetric (3D) kidney substructures segmentation from contrast-enhanced CT images (Arterial/Portal Venous Phase).
+
```
Comment (Nic-Ma) [inline models/renalStructures_UNEST_segmentation/docs/README.md:3]: I think this blank line is not necessary.

Thanks.

File: models/renalStructures_UNEST_segmentation/configs/inference.json (hunk 2 above)
Comment (Nic-Ma) [reply models/renalStructures_UNEST_segmentation/configs/inference.json:115]: I didn't see you updated the version in the `metadata.json`, did you push the commit?

Thanks.

File: models/wholeBrainSeg_Large_UNEST_segmentation/docs/README.md (hunk 5 above)
Comment (Nic-Ma) [reply models/wholeBrainSeg_Large_UNEST_segmentation/docs/README.md]: I see you still have the HTML code here?

Thanks.

File: models/renalStructures_UNEST_segmentation/docs/README.md (hunk 10 above)
Comment (tangy5) [reply models/renalStructures_UNEST_segmentation/docs/README.md:3]: Thanks, deleted.

File: models/renalStructures_UNEST_segmentation/docs/README.md (hunk 9 above)
Comment (tangy5) [reply models/renalStructures_UNEST_segmentation/docs/README.md]: Thanks, I'm not aware of these conflict lines. Checked for this file. 
Thanks again.

File: models/renalStructures_UNEST_segmentation/configs/inference.json (hunk 2 above)
Comment (tangy5) [reply models/renalStructures_UNEST_segmentation/configs/inference.json:115]:
```
    "changelog": {
        "0.1.0": "complete the model package",
        "0.1.1": "initialize the model package structure",
        "0.1.2": "Fixed the model state_dict name"
    },
```

It should be this section right? 
Yep, committed, thanks!

File: models/wholeBrainSeg_Large_UNEST_segmentation/docs/README.md [hunk 11]
```diff
@@ -0,0 +1,143 @@
...
+Training and testing data are MRI T1-weighted (T1w) 3D volumes coming from 3 different sites. There are a total of 133 classes in the whole brain segmentation task.
+Among 50 T1w MRI scans from Open Access Series on Imaging Studies (OASIS) (Marcus et al., 2007) dataset, 45 scans are used for training and the other 5 for validation.
+ The testing cohort contains Colin27 T1w scan (Aubert-Broche et al., 2006) and 13 T1w MRI scans from the Child and Adolescent Neuro Development Initiative (CANDI)
+ (Kennedy et al., 2012). All data are registered to the MNI space using the MNI305 (Evans et al., 1993) template and preprocessed follow the method in (Huo et al., 2019). Input images are randomly cropped to the size of 96 × 96 × 96.
+
+### Important
+
+```diff
++ All the brain MRI images for training are registered to Affine registration from the target image to the MNI305 template using NiftyReg.
++ The data should be in the MNI305 space before inference.
```
Comment (wyli) [inline models/wholeBrainSeg_Large_UNEST_segmentation/docs/README.md:51]: perhaps briefly mention the commands to achieve the registration?

File: models/renalStructures_UNEST_segmentation/configs/metadata.json [hunk 12]
```diff
@@ -3,7 +3,8 @@
     "version": "0.1.0",
```
Comment (yiheng-wang-nv) [inline models/renalStructures_UNEST_segmentation/configs/metadata.json]: Hi @tangy5 , here the version number should match the changelog

File: models/renalStructures_UNEST_segmentation/configs/metadata.json [hunk 13]
```diff
@@ -3,7 +3,8 @@
...
     "changelog": {
         "0.1.0": "complete the model package",
-        "0.0.1": "initialize the model package structure"
+        "0.1.1": "initialize the model package structure",
```
Comment (yiheng-wang-nv) [inline models/renalStructures_UNEST_segmentation/configs/metadata.json]: how about just remove this line, since we do not have an actual version `0.1.1` bundle here.

Comment (tangy5) [reply models/renalStructures_UNEST_segmentation/configs/metadata.json]: Ok, yes, this change is a bug fix. Let's remain as the same version, will delete this.

File: models/wholeBrainSeg_Large_UNEST_segmentation/docs/README.md (hunk 11 above)
Comment (tangy5) [reply models/wholeBrainSeg_Large_UNEST_segmentation/docs/README.md:51]: Yep, added a sample section for registration information. Thanks.

File: models/renalStructures_UNEST_segmentation/configs/metadata.json (hunk 12 above)
Comment (tangy5) [reply models/renalStructures_UNEST_segmentation/configs/metadata.json]: The version indeed cause confusion, let's remove it for now. If there are other versions in the future, will keep log change.

File: models/renalStructures_UNEST_segmentation/configs/metadata.json (hunk 13 above)
Comment (Nic-Ma) [reply models/renalStructures_UNEST_segmentation/configs/metadata.json]: Hi @tangy5 @yiheng-wang-nv ,

I think you should not remove the line, instead, please update the "version" to `0.1.1` and update the "changelog" to:
```
        "0.1.1": "fixed KeyError in the ChecpointLoader",
        "0.1.0": "complete the model package"
```
Refer to: https://github.com/Project-MONAI/model-zoo/blob/dev/models/brats_mri_segmentation/configs/metadata.json#L5
right?
What do you think?
Or maybe I misunderstand something?

Thanks.

Comment (yiheng-wang-nv) [reply models/renalStructures_UNEST_segmentation/configs/metadata.json]: Hi @Nic-Ma , the place you saw is outdated

File: models/renalStructures_UNEST_segmentation/configs/metadata.json (hunk 12 above)
Comment (yiheng-wang-nv) [reply models/renalStructures_UNEST_segmentation/configs/metadata.json]: No, "version" here should represent the latest version, in your case is "0.1.2"

File: models/renalStructures_UNEST_segmentation/configs/metadata.json (hunk 13 above)
Comment (yiheng-wang-nv) [reply models/renalStructures_UNEST_segmentation/configs/metadata.json]: Hi @Nic-Ma , 
> I think you should not remove the line

The place you saw is outdated, thus is not the place I mentioned that should be removed.

Comment (yiheng-wang-nv) [reply models/renalStructures_UNEST_segmentation/configs/metadata.json]: Hi @tangy5 , 
before changing this bundle, the meta information is:
```
"version": "0.1.0",
"changelog": {
    "0.1.0": "complete the model package",
    "0.0.1": "initialize the model package structure"
},
```

Then you made some changes, thus the version number should be updated, and mentioned in the "changelog". However, there should only be one item added into the changelog. Otherwise, you need to prepare two different versions of the bundle separately.

Therefore, during this PR you can write sth like:

```
"version": "0.1.1",
"changelog": {
    "0.1.1": "what you changed should be added here",
    "0.1.0": "complete the model package",
    "0.0.1": "initialize the model package structure"
},
```

Comment (tangy5) [reply models/renalStructures_UNEST_segmentation/configs/metadata.json]: This is very clear, thanks!
Fixed and commited

File: models/renalStructures_UNEST_segmentation/configs/metadata.json [hunk 14]
```diff
@@ -2,8 +2,7 @@
     "schema": "https://github.com/Project-MONAI/MONAI-extra-test-data/releases/download/0.8.1/meta_schema_20220324.json",
     "version": "0.1.0",
     "changelog": {
-        "0.1.0": "complete the model package",
-        "0.0.1": "initialize the model package structure"
+        "0.1.0": "complete the model package"
```
Comment (yiheng-wang-nv) [inline models/renalStructures_UNEST_segmentation/configs/metadata.json:6]: Hi @tangy5 , here the version number and changelog are still not changed. It should be updated into a new version.

File: models/wholeBrainSeg_Large_UNEST_segmentation/configs/metadata.json [hunk 15]
```diff
@@ -0,0 +1,211 @@
+{
+    "schema": "https://github.com/Project-MONAI/MONAI-extra-test-data/releases/download/0.8.1/meta_schema_20220324.json",
+    "version": "0.1.1",
+    "changelog": {
+        "0.1.0": "complete the model package",
+        "0.1.1": "fixed the model state dict name"
```
Comment (yiheng-wang-nv) [inline models/wholeBrainSeg_Large_UNEST_segmentation/configs/metadata.json]: Hi @tangy5 , it seems you may modify the wrong place, this bundle is a new one, thus it may not need to add two change log, maybe 0.1.1 can be removed.
In addition, usually we add the new changelog in the top of the dict.

Comment (tangy5) [reply models/wholeBrainSeg_Large_UNEST_segmentation/configs/metadata.json]: Fixed as below: 
```
"version": "0.1.1",
    "changelog": {
        "0.1.1": "fixed the model state dict name",
        "0.1.0": "complete the model package"
    },
```
Thanks

**Review by yiheng-wang-nv — APPROVED** [review]
